cmake/CheckPatch.cmake

```diff
@@ -0,0 +1,31 @@
+find_program(CHECKPATCH checkpatch.pl
+             HINTS ${PROJECT_SOURCE_DIR}/checkpatch)
+find_program(CODESPELL codespell)
+
+add_custom_target(checkpatch)
+
+set(BASE_GIT_REF "origin/master")
+if(DEFINED ENV{CHECKPATCH_GITREF})
+    set(BASE_GIT_REF "$ENV{CHECKPATCH_GITREF}")
+endif()
+
+# Description of supported checks in
+# https://github.com/tarantool/checkpatch/blob/master/doc/checkpatch.rst
+set(CHECKPATCH_OPTIONS --git ${BASE_GIT_REF}..HEAD --show-types)
+if(CODESPELL)
+    set(CHECKPATCH_OPTIONS ${CHECKPATCH_OPTIONS} --codespell)
+endif(CODESPELL)
+
+if(CHECKPATCH)
+    add_custom_command(TARGET checkpatch
+        COMMENT "Running checkpatch on ${CUR_GIT_REF} against ${BASE_GIT_REF}"
```

Comment:
`CUR_GIT_REF` is undefined (and not required here).

Reply:
Right, thanks!